plugins/genesys/fu-genesys-scaler-device.c

```diff
@@ -1417,17 +1551,28 @@ fu_genesys_scaler_device_probe(FuDevice *device, GError **error)
 
 	if (!fu_genesys_scaler_device_get_version(self, buf, sizeof(buf), error))
 		return FALSE;
-	version =
-	    fu_common_strsafe((const gchar *)&buf[1], 6); /* ?RIM123; where ? is 0x06 (length?) */
+	/* ?xIM123; where ? is 0x06 (length?) */
+	panel_type = fu_common_strsafe((const gchar *)&buf[1], 6);
+
+	if (!fu_genesys_scaler_device_get_firmware_packet_version(self, &ver, error))
+		return FALSE;
 
+	version = g_strdup_printf("%d.%d.%d.%d", ver.stage, ver.model, ver.major, ver.minor);
 	fu_device_set_version(device, version);
 	fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_PLAIN);
 	fu_device_set_logical_id(device, "scaler");
 
 	/* add instance ID */
 	fu_device_add_instance_str(device, "MSTAR", "TSUM_G");
 	fu_device_add_instance_strup(device, "PUBKEY", guid);
-	fu_device_build_instance_id(device, NULL, "GENESYS_SCALER", "MSTAR", "PUBKEY", NULL);
+	fu_device_add_instance_strup(device, "PANELTYPE", panel_type);
```

Comment:
maybe `PANEL`?

Reply:
This field refers to the attribute of `panel_type` in structure `FuGenesysMtkRsaHeader`; I like they have the same name if that possible. However, I like `PANEL`.

I would prefer `REV` or `HWREV`; but it is not clear to me if the that attribute always refers to a panel or not :/.

@adamgene, any opinion?

Reply:
>I would prefer `REV` or `HWREV`; but it is not clear to me if the that attribute always refers to a panel or not :/.
>

I researched the document for the `FuGenesysMtkRsaHeader`, it says `HW FW version`. But I found the string in HP definition is `REV`. The detail format can ask HP.

IMO, It is means panel revision for MSTAR scaler chip.

Reply:
REV won't work as we use that for the USB revision already.

Reply:
How about PANEL_REV?

Reply:
I like PANELREV more, but that works for me.

Reply:
`PANELREV` is works for me too.

Reply:
so let's go for `PANELREV`.